### PR TITLE
Add Slugged behavior

### PR DESCRIPTION
**Pull request by denisorehovsky**

### CODE REVIEW

File: behaviors/behaviors.py [hunk 1]
```diff
@@ -4,6 +4,8 @@
 from django.db import models
 from django.utils import timezone
 
+from slugify import slugify
+
```
Comment (audiolion) [inline behaviors/behaviors.py:8]: Is there a reason to add a dependency on awesome-slugify and not use `django.utils.text import slugify`? we are just lower casing the slug anyways which is django utils slugify default behavior

File: behaviors/behaviors.py [hunk 2]
```diff
@@ -98,6 +100,40 @@ def released(self):
...
+    An abstract behavior representing adding a unique slug to a model
+    based on the slug_source property.
+    """
+    slug = models.SlugField(max_length=255, unique=True)
+
+    class Meta:
+        abstract = True
+
+    def save(self, *args, **kwargs):
+        if not self.slug:
```
Comment (audiolion) [inline behaviors/behaviors.py:114]: We need to add a test case that covers the `True` condition of this, basically that the slug doesn't change when you re-save an object to not lose coverage

**Review by audiolion — COMMENTED** [review]
Thanks for the PR! I was meaning to get around to writing this, the implementation looks sound. The only real question I have is the usage of ``awesome-slugify`` over django's default slugify, let me know if there is any advantage of the external dependency.

File: behaviors/behaviors.py (hunk 1 above)
Comment (denisorehovsky) [reply behaviors/behaviors.py:8]: The reason is that standard django slugify doesn't work properly for non-english strings. https://www.pydanny.com/awesome-slugify-human-readable-url-slugs-from-any-string.html

File: behaviors/behaviors.py (hunk 2 above)
Comment (denisorehovsky) [reply behaviors/behaviors.py:114]: Yes, I can do that

File: behaviors/behaviors.py (hunk 1 above)
Comment (denisorehovsky) [reply behaviors/behaviors.py:8]: We can use unidecode package instead

Comment (audiolion) [reply behaviors/behaviors.py:8]: That seems like a good enough reason to me, we will just place a note in the docs about why we are using it, thanks!

File: behaviors/behaviors.py (hunk 2 above)
Comment (audiolion) [reply behaviors/behaviors.py:114]: I am going to update the docs and then we can merge this in!

Comment (audiolion) [reply behaviors/behaviors.py:114]: Sorry for the delay on this, I was moving and have had every night booked up cleaning, packing, moving, unpacking. Should be merged by this weekend.

Comment (denisorehovsky) [reply behaviors/behaviors.py:114]: It's okay. I thought that you forgot about this pull. :)

**Pull request merged**